Development version 1.16

raymondmiles opened the pull request:
This update uses the newest version of samtools (1.16.1) and incorporates changes in the dxapp.json as well as the readme.md.

In the dxapp.json:
Updated ubuntu instance to 20.04 from  16.04
Changed the name of the app to eggd_samtools_flagstat
Added the version field  v1.1.0
Added org-emee-1 to authorised users and developers
Updated instance type from mem1_ssd1_x2 to mem1_ssd1_v2_x2
Removed the github version field

In the readme.md:
Updated app description. 
Added an inputs section. 
Added list of outputted fields. 
Added examples on how to run on CLI.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)
<!-- Reviewable:end -->

chrispyatt reviewed (changes requested):
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 5 unresolved discussions (waiting on @mattgarner and @raymondmiles)

___
*[`dxapp.json` line 6 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMP711115P17sL01Cy:-NNMP711115P17sL01Cz:b-y7sbl4) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L6)):*
> ```JSON
>     "summary": "Provides counts for each of 16 categories based primarily on bit flags in the FLAG field. Each category in the output is broken down into QC pass and QC fail. In the default output format, these are presented as \"#PASS + #FAIL\" followed by a description of the category. See http://www.htslib.org/doc/samtools-flagstat.html for further documentation.",
>     "version": "1.1.0",
>     "whatsNew": "* v1.1.0 Updated name, version, instance type and ubuntu release, deleted githubrelease field.",
> ```

This could be clearer (does "version" refer to the app version or the samtools version?). Also missing the fact that samtools itself has been updated. Probably worth briefly mentioning what you've updated them to, e.g. samtools updated to v.1.16.1
___
*[`Readme.md` line 4 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMM1r20rUWGWOe-Une:-NNMM1r20rUWGWOe-Unf:bpicu1z) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L4)):*
> ```Markdown
> # eggd_samtools_flagstat (DNAnexus Platform App)
> 
> eggd_samtools_flagstat is an app used to quantify flags in the 
> ```

partial sentence
___
*[`Readme.md` line 11 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMLkSZ74xWpamJ7XYr:-NNMLkSZ74xWpamJ7XYs:br5q9dv) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L11)):*
> ```Markdown
> ## What does this app do?
> 
> This app runs the samtools flagstat script pn a .bam file and the corresponding .bam.bai index file and returns a .flagstat file.
> ```

typo - (pn instead of on)
___
*[`Readme.md` line 21 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMMBdcEqPNIUAnXuh0:-NNMMBdcEqPNIUAnXuh1:b-4n661y) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L21)):*
> ```Markdown
> 
> ## What are typical use cases for this app?
> This app should be executed stand-alone or as part of a DNAnexus workflow.
> ```

This should explain what we use the app for, i.e. calculating QC statistics on NGS data
___
*[`Readme.md` line 51 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMOEFf3q7Opsnvao79:-NNMOEFf3q7Opsnvao7A:b1p5lum) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L51)):*
> ```Markdown
> ## How to run this app from the command line?
> 
> dx run app-eggd_samtools_flagstat
> ```

I'd change this to be one command rather than copying the interactive method. I.e. `dx run app-eggd_samtools_flagstat -iinput_bam <fileID> -iinput_bam_index <fileID>` (check this is correct first as I've typed it off the top of my head)


<!-- Sent from Reviewable.io -->

mattgarner reviewed (changes requested):
> Reviewed 1 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 9 unresolved discussions (waiting on @chrispyatt and @raymondmiles)

___
*[`dxapp.json` line 6 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMP711115P17sL01Cy:-NNaRWFl6P2UJbTwoSRQ:bwcjj7g) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L6)):*
<details><summary><i>Previously, chrispyatt wrote…</i></summary><blockquote>

This could be clearer (does "version" refer to the app version or the samtools version?). Also missing the fact that samtools itself has been updated. Probably worth briefly mentioning what you've updated them to, e.g. samtools updated to v.1.16.1
</blockquote></details>

Agree. We wouldn't generally mention that we've updated the version of the app, since any update should come with a version number update so that's a given. However it would be good to be specific about the samtools update as Chris suggested.

Perhaps describe the most important changes first - the update to samtools and change from applet to app. Then the other minor details.
___
*[`dxapp.json` line 41 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaS6NFE6MnGF1rl8NU:-NNaS6NG3fWBP14i5mH_:b4p707c) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L41)):*
> ```JSON
>       }
>     ],
>     
> ```

Whitespace. Please remove the extra spaces.



_Code quote:_
```JSON
····
```
___
*[`dxapp.json` line 52 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaSgUq4eeUtszSlr73:-NNaSgUq4eeUtszSlr74:b8t5xa4) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L52)):*
> ```JSON
>       "distribution": "Ubuntu",
>       "file": "src/dnanexus_samtools_flagstat.sh",
>       "version": "1"
> ```

Is 1 the correct value to use here?
See https://documentation.dnanexus.com/developer/apps/execution-environment#choosing-an-application-execution-environment



_Code quote:_
```JSON
"1"
```
___
*[`dxapp.json` line 77 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaSuWk6UJf-XVs63_o:-NNaSuWk6UJf-XVs63_p:bdzzdcy) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L77)):*
> ```JSON
>     }
>   }
>   
> ```

Whitespace



_Code quote:_
```JSON
··
```
___
*[`Readme.md` line 13 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaTMftE170wjcUiH6c:-NNaTMftE170wjcUiH6d:undefined) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L13)):*
> ```Markdown
> This app runs the samtools flagstat script pn a .bam file and the corresponding .bam.bai index file and returns a .flagstat file.
> 
> The .flagstat file provides counts for each of 16 categories based primarily on bit flags in the FLAG field. Each category in the output is broken down into QC pass and QC fail. In the default output format, these are presented as "#PASS + #FAIL" followed by a description of the category. See http://www.htslib.org/doc/samtools-flagstat.html for further documentation. 
> ```





_Suggestion:_
```Markdown
read counts
```


<!-- Sent from Reviewable.io -->

mattgarner reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 10 unresolved discussions (waiting on @chrispyatt and @raymondmiles)

___
*[`Readme.md` line 33 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaTvCP3jdbEPsI5ykM:-NNaTvCQ042IwKfnmcs2:b-a6ypuf) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L33)):*
> ```Markdown
> #### Additional categories given for reads are:
> 
> 2. primary neither 0x100 nor 0x800 bit set 
> ```

Perhaps reformat these slightly as below for clarity



_Suggestion:_
```Markdown
primary (neither 0x100 nor 0x800 bit set)
```
___
*[`Readme.md` line 51 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMOEFf3q7Opsnvao79:-NNaU8ww79GVCodxZDmU:b4q2la5) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L51)):*
<details><summary><i>Previously, chrispyatt wrote…</i></summary><blockquote>

I'd change this to be one command rather than copying the interactive method. I.e. `dx run app-eggd_samtools_flagstat -iinput_bam <fileID> -iinput_bam_index <fileID>` (check this is correct first as I've typed it off the top of my head)
</blockquote></details>

Agree. If you've not used dx apps in this way feel free to ask any of us for guidance


<!-- Sent from Reviewable.io -->

chrispyatt reviewed (changes requested):
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 9 unresolved discussions (waiting on @raymondmiles)

___
*[`dxapp.json` line 6 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvISRj6u25p6wJpCMQ:-NNvISRk4Nj4oGCjKRmQ:b-xph2hg) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/dxapp.json#L6)):*
> ```JSON
>     "summary": "Provides read counts for each of 13 categories based primarily on bit flags in the FLAG field. Each category in the output is broken down into QC pass and QC fail. In the default output format, these are presented as \"#PASS + #FAIL\" followed by a description of the category. See http://www.htslib.org/doc/samtools-flagstat.html for further documentation",
>     "version": "1.1.0",
>     "whatsNew": "* v1.1.0 Updated name, instance type and ubuntu release, deleted githubrelease field. Updated samtools version to 1.16.1.",
> ```

As Matt says, move the "updated samtools" bit to the start as that's the major change.
___
*[`Readme.md` line 4 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNv6RGQ35ST4By_dnd4:-NNv6RGQ35ST4By_dnd5:b-h4u3pi) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/Readme.md#L4)):*
> ```Markdown
> # eggd_samtools_flagstat (DNAnexus Platform App)
> 
> eggd_samtools_flagstat is an app used to quantify the number of allignments for each FLAG field in a bam file. It takes in a bam and index file and calculates statistics based on bit flags in the FLAG field, information about the flags can be found in the following document: https://samtools.github.io/hts-specs/SAMv1.pdf
> ```

typo - alignments
___
*[`Readme.md` line 21 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvA-54-naiasfo5RH2:-NNvA-54-naiasfo5RH3:b-1nndoz) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/Readme.md#L21)):*
> ```Markdown
> 
> ## What are typical use cases for this app?
> This app should be executed to calculate Quality Control statics related to reads in .bam files.
> ```

typo - statistics


<!-- Sent from Reviewable.io -->

raymondmiles commented:
> > Reviewed 5 of 5 files at r1, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)_ status: all files reviewed, 5 unresolved discussions (waiting on @mattgarner and @raymondmiles)
> 
> _[`dxapp.json` line 6 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMP711115P17sL01Cy:-NNMP711115P17sL01Cz:b-y7sbl4) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L6)):_
> 
> > ```json
> >     "summary": "Provides counts for each of 16 categories based primarily on bit flags in the FLAG field. Each category in the output is broken down into QC pass and QC fail. In the default output format, these are presented as \"#PASS + #FAIL\" followed by a description of the category. See http://www.htslib.org/doc/samtools-flagstat.html for further documentation.",
> >     "version": "1.1.0",
> >     "whatsNew": "* v1.1.0 Updated name, version, instance type and ubuntu release, deleted githubrelease field.",
> > ```
> 
> This could be clearer (does "version" refer to the app version or the samtools version?). Also missing the fact that samtools itself has been updated. Probably worth briefly mentioning what you've updated them to, e.g. samtools updated to v.1.16.1
> 
> _[`Readme.md` line 4 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMM1r20rUWGWOe-Une:-NNMM1r20rUWGWOe-Unf:bpicu1z) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L4)):_
> 
> > ```gfm
> > # eggd_samtools_flagstat (DNAnexus Platform App)
> > 
> > eggd_samtools_flagstat is an app used to quantify flags in the 
> > ```
> 
> partial sentence
> 
> _[`Readme.md` line 11 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMLkSZ74xWpamJ7XYr:-NNMLkSZ74xWpamJ7XYs:br5q9dv) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L11)):_
> 
> > ```gfm
> > ## What does this app do?
> > 
> > This app runs the samtools flagstat script pn a .bam file and the corresponding .bam.bai index file and returns a .flagstat file.
> > ```
> 
> typo - (pn instead of on)
> 
> _[`Readme.md` line 21 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMMBdcEqPNIUAnXuh0:-NNMMBdcEqPNIUAnXuh1:b-4n661y) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L21)):_
> 
> > ```gfm
> > ## What are typical use cases for this app?
> > This app should be executed stand-alone or as part of a DNAnexus workflow.
> > ```
> 
> This should explain what we use the app for, i.e. calculating QC statistics on NGS data
> 
> _[`Readme.md` line 51 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMOEFf3q7Opsnvao79:-NNMOEFf3q7Opsnvao7A:b1p5lum) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L51)):_
> 
> > ```gfm
> > ## How to run this app from the command line?
> > 
> > dx run app-eggd_samtools_flagstat
> > ```
> 
> I'd change this to be one command rather than copying the interactive method. I.e. `dx run app-eggd_samtools_flagstat -iinput_bam <fileID> -iinput_bam_index <fileID>` (check this is correct first as I've typed it off the top of my head)



> > Reviewed 2 of 2 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)_ status: all files reviewed, 9 unresolved discussions (waiting on @raymondmiles)
> 
> _[`dxapp.json` line 6 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvISRj6u25p6wJpCMQ:-NNvISRk4Nj4oGCjKRmQ:b-xph2hg) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/dxapp.json#L6)):_
> 
> > ```json
> >     "summary": "Provides read counts for each of 13 categories based primarily on bit flags in the FLAG field. Each category in the output is broken down into QC pass and QC fail. In the default output format, these are presented as \"#PASS + #FAIL\" followed by a description of the category. See http://www.htslib.org/doc/samtools-flagstat.html for further documentation",
> >     "version": "1.1.0",
> >     "whatsNew": "* v1.1.0 Updated name, instance type and ubuntu release, deleted githubrelease field. Updated samtools version to 1.16.1.",
> > ```
> 
> As Matt says, move the "updated samtools" bit to the start as that's the major change.
> 
> _[`Readme.md` line 4 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNv6RGQ35ST4By_dnd4:-NNv6RGQ35ST4By_dnd5:b-h4u3pi) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/Readme.md#L4)):_
> 
> > ```gfm
> > # eggd_samtools_flagstat (DNAnexus Platform App)
> > 
> > eggd_samtools_flagstat is an app used to quantify the number of allignments for each FLAG field in a bam file. It takes in a bam and index file and calculates statistics based on bit flags in the FLAG field, information about the flags can be found in the following document: https://samtools.github.io/hts-specs/SAMv1.pdf
> > ```
> 
> typo - alignments
> 
> _[`Readme.md` line 21 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvA-54-naiasfo5RH2:-NNvA-54-naiasfo5RH3:b-1nndoz) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/Readme.md#L21)):_
> 
> > ```gfm
> > ## What are typical use cases for this app?
> > This app should be executed to calculate Quality Control statics related to reads in .bam files.
> > ```
> 
> typo - statistics

Added suggested changes to readme.md
    Typos
Added suggested changes to dxapp.json
    Moved updated samtools to the front of the What's new field

raymondmiles commented:
> Perhaps reformat these slightly as below for clarity


In the readme.md
    I added the braces to improve clarity. I am familiar with flags but I don't full understand this section of the documentation when it comes to terms like "set" and "neither"
    I added the CLI command in the readme regarding how to run the app.

raymondmiles reviewed (commented):
In the readme.md
Corrected typos.
I added the braces to the output section to improve clarity. I am familiar with flags but I don't full understand this section of the documentation when it comes to terms like "set" and "neither"
I added the CLI command in the readme regarding how to run the app.

In the dxapp.json
Incorporated suggested edits
Corrected runSpecVersion to 0
Typos

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-:-NNv_T5p4uZuv90HtYdv:b-y8ya9)* status: 3 of 5 files reviewed, 9 unresolved discussions (waiting on @chrispyatt and @mattgarner)

___
*[`dxapp.json` line 6 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMP711115P17sL01Cy:-NNvZLFQAWQXnI6Q5jnP:b-qh7svb) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L6)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Agree. We wouldn't generally mention that we've updated the version of the app, since any update should come with a version number update so that's a given. However it would be good to be specific about the samtools update as Chris suggested.

Perhaps describe the most important changes first - the update to samtools and change from applet to app. Then the other minor details.
</blockquote></details>

The version does refer to the app version so I removed it as it is in the version number update. I added in the version of samtools that is now being used and added a reference to the transition to an app.
___
*[`dxapp.json` line 41 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaS6NFE6MnGF1rl8NU:-NNv_4mB8dhm5HxVKzh1:b8ilsj9) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L41)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Whitespace. Please remove the extra spaces.
</blockquote></details>

Removed whitespaces in dxapp.json
___
*[`dxapp.json` line 52 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaSgUq4eeUtszSlr73:-NNv_8Fk574qkBiF73zg:bwjeln5) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L52)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Is 1 the correct value to use here?
See https://documentation.dnanexus.com/developer/apps/execution-environment#choosing-an-application-execution-environment
</blockquote></details>

No the correct value is a 0 cheers. I changed that back to 0.
___
*[`dxapp.json` line 77 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaSuWk6UJf-XVs63_o:-NNv_EuDDgcD1mSyVVM-:bss4a2b) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/dxapp.json#L77)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Whitespace
</blockquote></details>

Removed.
___
*[`dxapp.json` line 6 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvISRj6u25p6wJpCMQ:-NNvZ4_FDHSSg_sGNI2A:b-yvhx75) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/dxapp.json#L6)):*
>  Done
___
*[`Readme.md` line 13 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaTMftE170wjcUiH6c:-NNv_IvS2HhIAYJ8fkhp:b-1dcph8) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L13)):*
> ```Markdown
> This app runs the samtools flagstat script pn a .bam file and the corresponding .bam.bai index file and returns a .flagstat file.
> 
> The .flagstat file provides counts for each of 16 categories based primarily on bit flags in the FLAG field. Each category in the output is broken down into QC pass and QC fail. In the default output format, these are presented as "#PASS + #FAIL" followed by a description of the category. See http://www.htslib.org/doc/samtools-flagstat.html for further documentation. 
> ```

Corrected
___
*[`Readme.md` line 33 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaTvCP3jdbEPsI5ykM:-NNv_MEl5ElYDtI-6Av0:bbw75ov) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L33)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Perhaps reformat these slightly as below for clarity
</blockquote></details>

Added round brackets for clarity.
___
*[`Readme.md` line 4 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNv6RGQ35ST4By_dnd4:-NNv_GdD-YVN2Db5qczb:b-1dcph8) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/Readme.md#L4)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

typo - alignments
</blockquote></details>

Corrected
___
*[`Readme.md` line 21 at r2](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvA-54-naiasfo5RH2:-NNv_K9uCIAkJYIQXVtC:b-1dcph8) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/6bd77b8788da44d69aafc87650eb110eaf12fabc/Readme.md#L21)):*
<details><summary><i>Previously, raymondmiles (Raymond Cadoc Miles) wrote…</i></summary><blockquote>

Added suggested changes to readme.md
    Typos
Added suggested changes to dxapp.json
    Moved updated samtools to the front of the What's new field

</blockquote></details>

Corrected


<!-- Sent from Reviewable.io -->

mattgarner reviewed (changes requested):
> Reviewed 3 of 5 files at r1, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 7 unresolved discussions (waiting on @chrispyatt and @raymondmiles)

___
*[`Readme.md` line 13 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaTMftE170wjcUiH6c:-NNvleNxAKf98wR5ObfF:b-b829zy) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L13)):*
<details><summary><i>Previously, raymondmiles (Raymond Cadoc Miles) wrote…</i></summary><blockquote>

Corrected
</blockquote></details>

Looks like it still says `counts` rather than `read counts`
___
*[`Readme.md` line 46 at r4](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvm9fiCFXIJiHB6oBj:-NNvm9fiCFXIJiHB6oBk:b-iyvhz2) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/53c046e0e15401a236825a2000791a4c72437698/Readme.md#L46)):*
> ```Markdown
> 13. with itself and mate mapped (0x1 bit set and neither 0x4 nor 0x8 bits set)
> 14. singletons (both 0x1 and 0x8 bits set and bit 0x4 not set)
> ### And finally, two rows are given that additionally filter on the reference name (RNAME), mate reference name (MRNM), and mapping quality (MAPQ) fields:  
> ```

whitespace



_Code quote:_
```Markdown
··
```
___
*[`Readme.md` line 47 at r4](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvm7vkChE30m2mVlAk:-NNvm7vkChE30m2mVlAl:b-iyvhz2) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/53c046e0e15401a236825a2000791a4c72437698/Readme.md#L47)):*
> ```Markdown
> 14. singletons (both 0x1 and 0x8 bits set and bit 0x4 not set)
> ### And finally, two rows are given that additionally filter on the reference name (RNAME), mate reference name (MRNM), and mapping quality (MAPQ) fields:  
> 15. with mate mapped to a different chr     
> ```

whitespace



_Code quote:_
```Markdown
·····
```


<!-- Sent from Reviewable.io -->

raymondmiles reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-:-NNvn5IIAfoWRthzotXc:bnfp4nl)* status: 4 of 5 files reviewed, 7 unresolved discussions (waiting on @chrispyatt and @mattgarner)

___
*[`Readme.md` line 51 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNMOEFf3q7Opsnvao79:-NNvoeIA1GQZciu7q7nQ:b-14lcak) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L51)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Agree. If you've not used dx apps in this way feel free to ask any of us for guidance
</blockquote></details>

I changed this command to be a CLI instruction.
___
*[`Readme.md` line 46 at r4](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvm9fiCFXIJiHB6oBj:-NNvoSl96svIXSjHCSs2:bjtslmh) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/53c046e0e15401a236825a2000791a4c72437698/Readme.md#L46)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

whitespace
</blockquote></details>

Removed the whitespace
___
*[`Readme.md` line 47 at r4](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNvm7vkChE30m2mVlAk:-NNvoWdH59ah7kJvS5yO:b-omdsvo) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/53c046e0e15401a236825a2000791a4c72437698/Readme.md#L47)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

whitespace
</blockquote></details>

Removed the whitespaces.


<!-- Sent from Reviewable.io -->

raymondmiles reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 4 of 5 files reviewed, 7 unresolved discussions (waiting on @chrispyatt and @mattgarner)

___
*[`Readme.md` line 13 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNaTMftE170wjcUiH6c:-NNvokwBD5I72AegPOyr:b4q8ot9) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/Readme.md#L13)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Looks like it still says `counts` rather than `read counts`
</blockquote></details>

Apologies for that, I made the same mistake in the dxapp.json so when I corrected that I missed it. It should be fixed now.


<!-- Sent from Reviewable.io -->

mattgarner reviewed (approved):
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 4 unresolved discussions (waiting on @chrispyatt)



<!-- Sent from Reviewable.io -->

sophie22 reviewed (changes requested):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 9 unresolved discussions (waiting on @chrispyatt and @raymondmiles)

___
*[`dxapp.json` line 51 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNw9Rt6-SWTIfXeY85F:-NNw9Rt6-SWTIfXeY85G:bunk424) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L51)):*
> ```JSON
>       "distribution": "Ubuntu",
>       "file": "src/dnanexus_samtools_flagstat.sh",
>       "version": "0"
> ```

minor change request that for standardisation these could be rearranged as "distribution, release, version, interpreter and file". Following on from Rob's audit, there should be a "timeoutPolicy" field as well (as an example see https://github.com/eastgenomics/eggd_verifybamid/blob/master/dxapp.json#L49)



_Code quote:_
```JSON
      "interpreter": "bash",
      "release": "20.04",
      "distribution": "Ubuntu",
      "file": "src/dnanexus_samtools_flagstat.sh",
      "version": "0"
```
___
*[`dxapp.json` line 75 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwAApADIhrvMXkpl91:-NNwAApADIhrvMXkpl92:b8dwoe8) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L75)):*
> ```JSON
>       }
>     }
>   }
> ```

looks like this file doesn't have a newline character at the end, it's usually best to have an empty line before the end of file
___
*[`Readme.md` line 49 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwBxpj4Z6ABcPOogME:-NNwBxpj4Z6ABcPOogMF:b-1v3pu) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/Readme.md#L49)):*
> ```Markdown
> ## How to run this app from the command line?
> 
> dx run app-eggd_samtools_flagstat -iinput_bam <fileID> -iinput_bam_index <fileID>
> ```

these do not show up when not in editor view



_Code quote:_
```Markdown
<fileID>
```
___
*[`Readme.md` line 49 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwC2Mt09x7HuMKKdqb:-NNwC2Mt09x7HuMKKdqc:bnz85jz) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/Readme.md#L49)):*
> ```Markdown
> ## How to run this app from the command line?
> 
> dx run app-eggd_samtools_flagstat -iinput_bam <fileID> -iinput_bam_index <fileID>
> ```

I recommend placing the command between ```signs``` for it to be formatted as code.



_Code quote:_
```Markdown
dx run app-eggd_samtools_flagstat -iinput_bam <fileID> -iinput_bam_index <fileID>
```
___
*[`Readme.md` line 50 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwAykr3atnuO_srNb9:-NNwAykr3atnuO_srNbA:b-91nqlj) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/Readme.md#L50)):*
> ```Markdown
> 
> dx run app-eggd_samtools_flagstat -iinput_bam <fileID> -iinput_bam_index <fileID>
> ### This app was made by EMEE GLH
> ```

here as well, better to have a newline before the end of file


<!-- Sent from Reviewable.io -->

raymondmiles reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 2 of 5 files reviewed, 13 unresolved discussions (waiting on @chrispyatt, @jethror1, @mattgarner, and @sophie22)

___
*[`dxapp.json` line 51 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNw9Rt6-SWTIfXeY85F:-NO-UHV44_23nQz893ww:byv9doa) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L51)):*
<details><summary><i>Previously, sophie22 (Sophie Ratkai) wrote…</i></summary><blockquote>

minor change request that for standardisation these could be rearranged as "distribution, release, version, interpreter and file". Following on from Rob's audit, there should be a "timeoutPolicy" field as well (as an example see https://github.com/eastgenomics/eggd_verifybamid/blob/master/dxapp.json#L49)
</blockquote></details>

Will do, there is a timeout policy set to 2 hours just before this block of code 
![image.png](https://files.reviewableusercontent.io/images/reviewable/17769948/JLiHT_rYaiLVk4UIjdDDVS1T/image.png)
___
*[`dxapp.json` line 62 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwIDKkBhFMKARTZAKv:-NO-V2I-C4oz8RmPdAnJ:b-lirtsm) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L62)):*
> Having no network access would mitigate this as it would block outbound traffic to any endpoints that it might connect back to, and our practice is to not build apps reliant on externally pulled resources at run time anyway.
How would I mitigate this? Would I remove the network field entirely?
___
*[`dxapp.json` line 75 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwAApADIhrvMXkpl91:-NO-VNq73jrS80bToPT0:b-day9j4) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L75)):*
<details><summary><i>Previously, sophie22 (Sophie Ratkai) wrote…</i></summary><blockquote>

looks like this file doesn't have a newline character at the end, it's usually best to have an empty line before the end of file
</blockquote></details>

Would it be a new line without a whitespace? 
___
*[`Readme.md` line 49 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwC2Mt09x7HuMKKdqb:-NO-WhAk47MAGA2fpm-D:bf190ls) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/Readme.md#L49)):*
<details><summary><i>Previously, sophie22 (Sophie Ratkai) wrote…</i></summary><blockquote>

I recommend placing the command between ```signs``` for it to be formatted as code.
</blockquote></details>

I am unfamiliar with signs however I used backticks ``` before and after the code following https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks
To visualise the code better. 
___
*[`Readme.md` line 50 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwAykr3atnuO_srNb9:-NO-XJ4fCDSUQ4YBO-Mz:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/Readme.md#L50)):*
<details><summary><i>Previously, sophie22 (Sophie Ratkai) wrote…</i></summary><blockquote>

here as well, better to have a newline before the end of file
</blockquote></details>

Done.
___
*[`src/dnanexus_samtools_flagstat.sh` line 8 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwHtSH18VeCGHpNQEB:-NO-Y5u8B2cndH8P_NNw:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/src/dnanexus_samtools_flagstat.sh#L8)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

all of this can be removed, its just default comments from the app wizard
</blockquote></details>

Done.
___
*[`src/dnanexus_samtools_flagstat.sh` line 59 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwI-wT8kwUkaZmpewQ:-NO-Y5Z56PiyULRgWe4e:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/src/dnanexus_samtools_flagstat.sh#L59)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

also can be deleted
</blockquote></details>

Done.
___
*[`src/dnanexus_samtools_flagstat.sh` line 73 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwI1YJBdOTUwBlXqRc:-NO-Y5Dt2wAADJbJkgmv:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/src/dnanexus_samtools_flagstat.sh#L73)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

and delete
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

sophie22 reviewed (changes requested):
> Reviewed 2 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 4 of 5 files reviewed, 10 unresolved discussions (waiting on @chrispyatt, @jethror1, @mattgarner, and @raymondmiles)

___
*[`dxapp.json` line 51 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNw9Rt6-SWTIfXeY85F:-NO9yQEA3ihUlO78cPxK:btw58lv) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L51)):*
<details><summary><i>Previously, raymondmiles (Raymond Cadoc Miles) wrote…</i></summary><blockquote>

Will do, there is a timeout policy set to 2 hours just before this block of code 
![image.png](https://files.reviewableusercontent.io/images/reviewable/17769948/JLiHT_rYaiLVk4UIjdDDVS1T/image.png)

</blockquote></details>

Thanks for making the changes to the order and pointing out that timeout has been included.
___
*[`dxapp.json` line 75 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwAApADIhrvMXkpl91:-NO9ykLDCqTZ4ORXRO2P:b-jp377f) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L75)):*
<details><summary><i>Previously, raymondmiles (Raymond Cadoc Miles) wrote…</i></summary><blockquote>

Would it be a new line without a whitespace? 
</blockquote></details>

correct
___
*[`Readme.md` line 49 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwC2Mt09x7HuMKKdqb:-NO9zBoy3blD4HQBSfxK:buumbfg) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/Readme.md#L49)):*
<details><summary><i>Previously, raymondmiles (Raymond Cadoc Miles) wrote…</i></summary><blockquote>

I am unfamiliar with signs however I used backticks ``` before and after the code following https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks
To visualise the code better. 
</blockquote></details>

I didn't realise it would make my backtick characters disappear, thanks for figuring out how to display code with the correct markup. This now looks great!


<!-- Sent from Reviewable.io -->

raymondmiles reviewed (commented):
Acknowledgements

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-:-NOA2g3b4fGSCX09L5JG:b62deji)* status: 4 of 5 files reviewed, 10 unresolved discussions (waiting on @chrispyatt, @jethror1, and @mattgarner)



<!-- Sent from Reviewable.io -->

chrispyatt reviewed (changes requested):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 4 of 5 files reviewed, 7 unresolved discussions (waiting on @jethror1, @mattgarner, and @raymondmiles)

___
*[`src/dnanexus_samtools_flagstat.sh` line 50 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NOKCc_y6JQMnQDBOcwQ:-NOKCc_y6JQMnQDBOcwR:bq5wd8l) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/src/dnanexus_samtools_flagstat.sh#L50)):*
> ```Shell
> 
>     outfile=${input_bam_prefix}.flagstat
>     samtools flagstat input_bam > $outfile
> ```

have you deleted the actual flagstat call here?


<!-- Sent from Reviewable.io -->

raymondmiles reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 4 of 5 files reviewed, 7 unresolved discussions (waiting on @chrispyatt, @jethror1, and @mattgarner)

___
*[`src/dnanexus_samtools_flagstat.sh` line 50 at r1](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NOKCc_y6JQMnQDBOcwQ:-NOPLYcC4yP7jvtKyxs0:b-dxo1ow) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/9ecdb311a01520f139ef221cf537d4e0693664d4/src/dnanexus_samtools_flagstat.sh#L50)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

have you deleted the actual flagstat call here?
</blockquote></details>

That was a massive mistake thanks for spotting! Now its back.


<!-- Sent from Reviewable.io -->

raymondmiles reviewed (commented):
> Reviewed 1 of 3 files at r6.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 4 of 5 files reviewed, 6 unresolved discussions (waiting on @chrispyatt and @jethror1)



<!-- Sent from Reviewable.io -->

raymondmiles reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 4 of 5 files reviewed, 6 unresolved discussions (waiting on @chrispyatt, @jethror1, and @sophie22)

___
*[`dxapp.json` line 75 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwAApADIhrvMXkpl91:-NOPM-rwCEdWRmqmJOQE:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L75)):*
<details><summary><i>Previously, sophie22 (Sophie Ratkai) wrote…</i></summary><blockquote>

correct
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

sophie22 reviewed (changes requested):
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 8 unresolved discussions (waiting on @chrispyatt, @jethror1, and @raymondmiles)

___
*[`src/dnanexus_samtools_flagstat.sh` line 18 at r7](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NOPW2k5-gqZe7SYvxMU:-NOPW2k5-gqZe7SYvxMV:b-s4qrvi) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/1e245fb39b6a517606d94471d22df7f9599257aa/src/dnanexus_samtools_flagstat.sh#L18)):*
> ```Shell
>     make install
>     export PATH=/packages/bin:$PATH
>     
> ```

whitespaces can be removed



_Code quote:_
```Shell
····
```
___
*[`src/dnanexus_samtools_flagstat.sh` line 32 at r7](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NOPWOUT5l6S3DGfYkVW:-NOPWOUT5l6S3DGfYkVX:bcu1c9m) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/1e245fb39b6a517606d94471d22df7f9599257aa/src/dnanexus_samtools_flagstat.sh#L32)):*
> ```Shell
> 
>     dx-jobutil-add-output flagstat_output "$flagstat_output" --class=file
> }
> ```

add an empty line before the end of file



_Code quote:_
```Shell
}
```


<!-- Sent from Reviewable.io -->

sophie22 commented:
fixes #3

raymondmiles reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: 4 of 5 files reviewed, 8 unresolved discussions (waiting on @chrispyatt, @jethror1, and @sophie22)

___
*[`src/dnanexus_samtools_flagstat.sh` line 18 at r7](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NOPW2k5-gqZe7SYvxMU:-NOQ6kHe1LAk5EPAr-6z:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/1e245fb39b6a517606d94471d22df7f9599257aa/src/dnanexus_samtools_flagstat.sh#L18)):*
<details><summary><i>Previously, sophie22 (Sophie Ratkai) wrote…</i></summary><blockquote>

whitespaces can be removed
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

sophie22 reviewed (changes requested):
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 7 unresolved discussions (waiting on @chrispyatt, @jethror1, and @raymondmiles)

___
*[`src/dnanexus_samtools_flagstat.sh` line 32 at r7](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NOPWOUT5l6S3DGfYkVW:-NOQHUM60BCR3sLjIx8O:b-q33o5x) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/1e245fb39b6a517606d94471d22df7f9599257aa/src/dnanexus_samtools_flagstat.sh#L32)):*
<details><summary><i>Previously, sophie22 (Sophie Ratkai) wrote…</i></summary><blockquote>

add an empty line before the end of file
</blockquote></details>

I meant to have it after the } closing bracket sign


<!-- Sent from Reviewable.io -->

jethror1 reviewed (approved):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 3 unresolved discussions (waiting on @raymondmiles and @sophie22)

___
*[`dxapp.json` line 62 at r5](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6#-NNwIDKkBhFMKARTZAKv:-NOUl9Br9Mj8PiyiAONy:bgn2k93) ([raw file](https://github.com/eastgenomics/eggd_samtools_flagstat/blob/f3283da1de98f8ad4f3804095726f59deb53d41a/dxapp.json#L62)):*
<details><summary><i>Previously, raymondmiles (Raymond Cadoc Miles) wrote…</i></summary><blockquote>

> Having no network access would mitigate this as it would block outbound traffic to any endpoints that it might connect back to, and our practice is to not build apps reliant on externally pulled resources at run time anyway.
How would I mitigate this? Would I remove the network field entirely?
</blockquote></details>

yes but don't worry about this for now, will discuss later within the team aside from this PR


<!-- Sent from Reviewable.io -->

jethror1 reviewed (approved):
> Reviewed 1 of 3 files at r6, 1 of 1 files at r8.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 3 unresolved discussions (waiting on @raymondmiles and @sophie22)



<!-- Sent from Reviewable.io -->

sophie22 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_samtools_flagstat/6)* status: all files reviewed, 1 unresolved discussion (waiting on @raymondmiles)



<!-- Sent from Reviewable.io -->
